### PR TITLE
[BEAM-3557] Sets parent pointer of AppliedPTransform objects correctly

### PR DESCRIPTION
**Pull request by chamikaramj**
Sets parent pointer correctly when constructing AppliedPTransform objects from a runner API proto.

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @charlesccychen

File: sdks/python/apache_beam/pipeline.py [hunk 1]
```diff
@@ -788,8 +788,11 @@ def is_side_input(tag):
       for si, pcoll in zip(result.transform.side_inputs, side_inputs):
         si.pvalue = pcoll
       result.side_inputs = tuple(result.transform.side_inputs)
-    result.parts = [
-        context.transforms.get_by_id(id) for id in proto.subtransforms]
+    result.parts = []
+    for id in proto.subtransforms:
```
Comment (charlesccychen) [inline sdks/python/apache_beam/pipeline.py]: Please use a different variable name (maybe `transform_id`)?  Python has `id` as a built-in function.

Comment (chamikaramj) [reply sdks/python/apache_beam/pipeline.py]: Done.

**Comment by chamikaramj** [comment]
Thanks.

**Comment by charlesccychen** [comment]
Thanks, LGTM.

**Pull request merged**